### PR TITLE
feat(ci): add e2e tests for connector sink and integrate sink e2e tests to ci

### PR DESCRIPTION
**Pull request by KivenChen**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

This PR integrates e2e tests for sink in PR & main branch workflows, which covers:
- mysql sink (already there, not integrated)
- connector sink via jdbc to postgres (newly added)
- create-as-sink queries
- any future e2e tests that matches (`e2e/sink/*.slt`)

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

File: ci/scripts/e2e-sink-test.sh [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+apt-get -y install mysql-client
+mysql --host=mysql --port=3306 -u root -p123456 -e "CREATE DATABASE IF NOT EXISTS test;"
+
+echo "--- installing postgresql"
+
+# set up PG sink destination
+ln -fs /usr/share/zoneinfo/America/New_York /etc/localtime
+DEBIAN_FRONTEND=noninteractive apt-get install -y tzdata
+dpkg-reconfigure --frontend noninteractive tzdata
+apt-get -y install postgresql postgresql-contrib
```
Comment (tabVersion) [inline ci/scripts/e2e-sink-test.sh:58]: can we install the dep when building docker?

**Review by tabVersion — APPROVED** [review]
generally LGTM

File: e2e_test/sink/t2.slt [hunk 2]
```diff
@@ -9,6 +9,7 @@ CREATE SINK s2 FROM mv2 WITH (
     connector = 'mysql',
     endpoint = '127.0.0.1:3306',
     user = 'root',
+    password = '123456'
     database = 'test',
     table = 't2'
```
Comment (tabVersion) [inline e2e_test/sink/t2.slt:14]: we are testing with sink node, so should set `connector = 'jdbc'` here?

Comment (KivenChen) [reply e2e_test/sink/t2.slt:14]: `t2.slt` and `t3.slt` are mysql sinks that exists long before this PR.

File: ci/scripts/e2e-sink-test.sh (hunk 1 above)
Comment (KivenChen) [reply ci/scripts/e2e-sink-test.sh:58]: Yep that should be better

File: ci/scripts/e2e-sink-test.sh [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+# start risingwave cluster
+cargo make ci-start ci-1cn-1fe
+
+# start connector node service
+nohup java -jar ./connector-service.jar > .risingwave/log/connector-source.log 2>&1 &
+sleep 1
+
+# test remote sink
+sqllogictest -p 4566 -d dev './e2e_test/sink/*.slt'
+sqllogictest -p 4566 -d dev './e2e_test/sink/remote/remote.load.slt'
```
Comment (wenym1) [inline ci/scripts/e2e-sink-test.sh:70]: Seems that the remote sink slt file is not included in this PR?

**Comment by KivenChen** [comment]
Ready for a final review after all fixes cc @wenym1

**Comment by KivenChen** [comment]
Added a fix for #6648 which wasn't exposed by ci, because sink e2e tests were just put there and not integrated into ci at all...

**Review by wenym1 — APPROVED** [review]
LGTM!

**Pull request merged**